course/include/Isometry.hpp

```diff
@@ -151,5 +158,75 @@ inline std::ostream& operator<<(std::ostream& os, const Matrix3& mat) {
               << "]]";
 }
 
+const inline Matrix3 matrix_multiplication(Matrix3 mat, Matrix3 rhs) {
+    Matrix3 result;
+    for (uint16_t row = 0; row < 3; ++row) {
+        for (uint16_t column = 0; column < 3; ++column) {
+            result[row][column] =
+                mat[row][0] * rhs[0][column] + mat[row][1] * rhs[1][column] + mat[row][2] * rhs[2][column];
+        }
+    }
+    return result;
+}
+
+class Isometry {
+   public:
+    Isometry(const Matrix3& rotation);
+    Isometry(const Vector3& translation, const Matrix3& rotation);
+
+    Isometry compose(const Isometry& isometry) const;
+    Isometry inverse() const;
+    Matrix3 rotation() const;
+    Vector3 transform(std::initializer_list<double> values) const;
```

Comment:
The argument to this would be a `Vector3`, since those are `x, y, z` coordinates.